### DISCUSSION TITLE
Using Git commit SHA for docker image tagging, how does rennovatebot decide which tag to use?

### DISCUSSION THREAD

**Discussion by hawkesn** [renovatebot/renovate#23459] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We are currently using Git commit SHA (not digest) to tag our images and then upload them to ECR (AWS Elastic Container Registry). 
On AWS ECR, we've turned on tag immutability (prevents uploads to the same tag).

We want rennovatebot to grab the "latest" commit (not tagged as `latest`, but the last image that was uploaded chronologically). 

I am currently using the following `packageRules`:
```
    {
      "matchPackagePrefixes": ["<aws-account-id>.dkr.ecr.<aws-region>.amazonaws.com"],
      "matchDatasources": ["docker"],
      "versioning": "regex:^(?<patch>[0-9a-fA-F]+$)"
    }
```
Where the regex is "pretending" that the Git commit SHA is a "patch" - (we eventually want to have this automerge)

Given that all tags in the registry are technically "fair game" for the replacement string, how does rennovatebot decide which tag to use?

When running this, the patches suggested are sometimes the latest, sometimes the 2nd latest, sometimes completely out of date... How can I modify rennovatebot (or the versioning) so that I can select it based on the latest chronological image?

### What you tried so far.

Modifying the versioning to `loose`, trying different regex patterns, trying to use `regexManager` instead.

This approach works with argocd-image-updater set to use `latest`
`argocd-image-updater.argoproj.io/xxxxxx.update-strategy: latest`
https://argocd-image-updater.readthedocs.io/en/stable/basics/update-strategies/#strategy-latest

But we'd rather prefer to use a single solution for everything rather than two

**Comment by hawkesn** [comment]
Taking a look at the [Docker Datasource](https://docs.renovatebot.com/modules/datasource/docker/) - it says: 
```
This datasource looks for the metadata of the latest stable image found on the Docker registry
```
So this should be loading the "latest"

**Comment by viceice** [comment, marked as answer]
it's not possible for renovate to follow git sha as versions because they don't have an order.

if you have a latest tag, then renovate can update the digest for that image when latest tag changes.

for renovate to update to different docker tags, there must be Al least one an increasing number for ordering.
